### PR TITLE
[test_pretest] Checks the state of each feature.

### PR DESCRIPTION
**Pull request by yozhao101**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
This PR aims to check whether the state of each feature in `CONFIG_DB` is valid or not. 

#### How did you do it?
I added a function in `test_pretest.py` to do continuously check the state of each feature.

#### How did you verify/test it?
I verified this change on DuT `str-msn2700-03`.

#### Any platform specific information?
N/A

#### Supported testbed topology if it's a new test case?
N/A

### CODE REVIEW

**Review by wangxin — APPROVED** [review]

File: tests/common/helpers/dut_utils.py [hunk 1]
```diff
@@ -230,3 +230,33 @@ def decode_dut_and_container_name(name_str):
...
+    """
+    feature_status, succeeded = duthost.get_feature_status()
+    if not succeeded:
+        logger.info("Failed to get list of feature names.")
+        return False
+
+    for feature_name, status in feature_status.items():
+        get_command = 'sonic-db-cli CONFIG_DB HGET "FEATURE|{}" state'.format(feature_name)
+        get_command_result = duthost.shell(get_command)["stdout_lines"][0]
+        feature_state = get_command_result.strip()
```
Comment (yxieca) [inline tests/common/helpers/dut_utils.py:252]: Why getting state again when you already have it in status? Why not just check status?

Comment (yozhao101) [reply tests/common/helpers/dut_utils.py:252]: I should reuse the `status` at here and will update.

**Comment by bingwang-ms** [comment]
PR testing for this one failed
```
        pytest_assert(wait_until(FEATURE_STATE_VERIFYING_THRESHOLD_SECS, FEATURE_STATE_VERIFYING_INTERVAL_SECS,
>                                verify_features_state, duthost), "Not all service states are valid!")
E       Failed: Not all service states are valid!
```
Is it because 10 min is not enough?

**Comment by yozhao101** [comment]
> PR testing for this one failed
> 
> ```
>         pytest_assert(wait_until(FEATURE_STATE_VERIFYING_THRESHOLD_SECS, FEATURE_STATE_VERIFYING_INTERVAL_SECS,
> >                                verify_features_state, duthost), "Not all service states are valid!")
> E       Failed: Not all service states are valid!
> ```
> 
> Is it because 10 min is not enough?

The `hostcfgd` only delays 90 seconds to start. We wait for 10 minutes and that should be fine.

**Comment by bingwang-ms** [comment]
It's still not rendered after 10 min
```
"dhcp_relay      {% if not (DEVICE_METADATA is defined and DEVICE_METADATA['localhost'] is defined and DEVICE_METADATA['localhost']['type'] is defined and DEVICE_METADATA['localhost']['type'] != 'ToRRouter') %}enabled{% else %}disabled{% endif %}  disabled        local",
```
Perhaps not a timing issue?

File: tests/common/helpers/dut_utils.py [hunk 2]
```diff
@@ -230,3 +232,30 @@ def decode_dut_and_container_name(name_str):
...
+    feature_status, succeeded = duthost.get_feature_status()
+    if not succeeded:
+        logger.info("Failed to get list of feature names.")
+        return False
+
+    for feature_name, status in feature_status.items():
+        logger.info("The state of '{}' is '{}'.".format(feature_name, status))
+
+        if status not in ("enabled", "always_enabled", "disabled", "always_disabled"):
+            logger.info("The state of '{}' is invalid!".format(feature_name))
```
Comment (lguohan) [inline tests/common/helpers/dut_utils.py:256]: please put more information in the log for debugging and troubleshoot, for example, what is the status now?

Comment (yozhao101) [reply tests/common/helpers/dut_utils.py:256]: The statement at line 253 will put the current `state` into the log.

**Review by yxieca — APPROVED** [review]

**Pull request merged**